tests/staking/StakingTest.js

```diff
@@ -1595,7 +1587,165 @@ contract("Staking", (accounts) => {
+
+            // TODO: should not require stake
+            const blockNumber = await initializeStake(date, "1000");
+
+            await expect(
+                staking.weightedStakeByDate(a1, date, startDate, blockNumber)
+            ).to.be.revertedWith("remaining time > max duration");
+        });
+
+        it("if blockNumber >= the current block number, the function reverts", async () => {
+            const currentBlockNumber = await web3.eth.getBlockNumber();
+            const date = kickoffTS.add(TWO_WEEKS_BN);
+            await expect(
+                staking.weightedStakeByDate(a1, date, date, currentBlockNumber)
+            ).to.be.revertedWith("not determined");
+            await expect(
+                staking.weightedStakeByDate(a1, date, date, currentBlockNumber + 1)
+            ).to.be.revertedWith("not determined");
+        });
+
+        it("returns the correct weight * stake for account and date (max duration)", async () => {
+            const startDate = kickoffTS;
+            // NOTE: this must be exactly the max lock date, or else `stake` will adjust the locked ts
+            const date = startDate.add(MAX_DURATION);
+            const blockNumber = await initializeStake(date, "10000");
+
+            const weightedStake = await staking.weightedStakeByDate(
+                a1,
+                date,
+                startDate,
+                blockNumber
+            );
+            expect(weightedStake).to.be.bignumber.eq(
+                getAmountWithWeight("10000", date, startDate)
+            );
+        });
+
+        it("returns the correct weight * stake for account and date (min duration)", async () => {
+            const startDate = kickoffTS.add(TWO_WEEKS_BN);
+            const date = startDate;
+            const blockNumber = await initializeStake(date, "10000");
+
+            const weightedStake = await staking.weightedStakeByDate(
+                a1,
+                date,
+                startDate,
+                blockNumber
+            );
+            expect(weightedStake).to.be.bignumber.eq(
+                getAmountWithWeight("10000", date, startDate)
+            );
+        });
+
+        it("returns the correct weight * stake for account and date (middle duration)", async () => {
+            const startDate = kickoffTS.add(TWO_WEEKS_BN);
+            const date = startDate.add(MAX_DURATION.div(new BN(2)));
+            const blockNumber = await initializeStake(date, "10000");
+
+            const weightedStake = await staking.weightedStakeByDate(
+                a1,
+                date,
+                startDate,
+                blockNumber
+            );
+            expect(weightedStake).to.be.bignumber.eq(
+                getAmountWithWeight("10000", date, startDate)
+            );
+        });
+
+        xit("if date is not a valid lock date, the function will return the weighted stake at the closest lock date prior to date", async () => {
```

Comment:
the spec of SOV-1216 says 
`if date is not a valid lock date, the function will return the weighted stake at the closest lock date AFTER date`